packages/react-devtools-shared/src/devtools/views/Components/InspectedElementErrorsAndWarningsTree.js

```diff
@@ -0,0 +1,146 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @flow
+ */
+
+import * as React from 'react';
+import Button from '../Button';
+import ButtonIcon from '../ButtonIcon';
+import Store from '../../store';
+import sharedStyles from './InspectedElementSharedStyles.css';
+import styles from './InspectedElementErrorsAndWarningsTree.css';
+import {SettingsContext} from '../Settings/SettingsContext';
+
+import type {InspectedElement} from './types';
+import type {FrontendBridge} from 'react-devtools-shared/src/bridge';
+
+type Props = {|
+  bridge: FrontendBridge,
+  inspectedElement: InspectedElement,
+  store: Store,
+|};
+
+export default function InspectedElementErrorsAndWarningsTree({
+  bridge,
+  inspectedElement,
+  store,
+}: Props) {
+  const {showInlineWarningsAndErrors} = React.useContext(SettingsContext);
+  if (!showInlineWarningsAndErrors) {
+    return null;
+  }
+
+  const {errors, warnings} = inspectedElement;
+
+  // TODO Would be nice if there were some way to either:
+  // (1) Temporarily disable the button after click (unstable_useTransition?) or
+  // (2) Immediately reflect the cleared list of warnings/errors.
+  // The current clear button feels a little unresponsive because we wait to poll for new values.
```

Comment:
Resolve in this PR or as a follow-up?

Reply:
Undecided. Need to review TODOs one more time.